File: 이상욱/BOJ/step/BOJ11399.java
package boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BOJ11399 {
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int num = Integer.parseInt(st.nextToken());

		int[] arr = new int[num];

		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < num; i++) {

			arr[i] = Integer.parseInt(st.nextToken());
		}

		Arrays.sort(arr);

		for (int i = 1; i < arr.length; i++) {
			arr[i] += arr[i - 1];
		}

		int sum = 0;
		for (int i = 0; i < arr.length; i++) {
			sum += arr[i];
		}

		System.out.println(sum);

	}
}

File: 이상욱/BOJ/step/BOJ11399.kt
package level16

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*


fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))
    val num = br.readLine().toInt()

    val arr = IntArray(num)
    val sum = IntArray(num)

    val st = StringTokenizer(br.readLine())

    for (i in 0 until num){

        arr[i] = st.nextToken().toInt()

    }

    Arrays.sort(arr)

    sum[0] = arr[0]
    var res = arr[0]

    for (i in 1 until  arr.size){
        sum[i] = arr[i] + sum[i-1]
        res += sum [i]
    }

    println(res)

}